src/nodes.coffee

```diff
@@ -1187,6 +1197,28 @@ exports.Obj = class Obj extends Base
       prop = prop.unwrapAll()
       prop.eachName iterator if prop.eachName?
 
+  # Object spread properties. https://github.com/tc39/proposal-object-rest-spread/blob/master/Spread.md
+  # obj2 = {a:1, obj..., c:3, d:4} => obj2 = Object.assign({}, {a:1}, obj1, {c:3, d:4})
+  compileSpread: (o) ->
+    props = @properties
+    # Store object spreads.
+    splatSlice = []
+    propSlices = []
+    slices = [new Obj [], false]
```

Comment:
Both of these parameters are optional and default to `[]` and `false` respectively, so this could just be `[new Obj]`.

Also, bit of a bigger one, but perhaps instead of starting with an empty object we could `.unshift` it at the end `unless slices[0] instanceof Obj` - that would save an empty object in case there are keys before splats (e.g. `{a, b...}` would be `Object.assign({ a }, b)` instead of `Object.assign({}, { a }, b)`).